waver/_dock_widget.py

```diff
@@ -0,0 +1,43 @@
+from magicgui import magic_factory
+from napari_plugin_engine import napari_hook_implementation
+from .simulation import Simulation
+
+
+@magic_factory(call_button="run",
+              spacing={'min': 1e-6, 'max':1},
+              speed={'min': 1, 'max':1e5},
+              time_step={'min': 1e-10, 'max':1},
+)
+def simulation(spacing: float=1e-3,
+               speed: float=343, 
+               time_step: float=200e-9) -> 'napari.types.LayerDataTuple':
+    """Run a single simulation.
+    """
+    # Should be provided by magicgui, right now hard coded
+    size = (12.8e-3, 12.8e-3) # how to do tuple - could be 1D, 2D, 3D?
```

Comment:
Not sure how to provide this tuple through magicgui, could be 1D/ 2D/ or 3D. Might need a more custom widget. Or maybe `ndim`, and `length` will work if i just make square/ cubes etc.